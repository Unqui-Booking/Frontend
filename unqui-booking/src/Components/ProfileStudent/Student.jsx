import { Avatar, Card, CardContent, Container, Grid, IconButton, Typography } from '@material-ui/core';
import React, { useEffect }  from 'react'
import { connect } from 'react-redux';
import moment from 'moment';
import { makeStyles } from '@material-ui/core/styles';
import { FaUser } from 'react-icons/fa';
import DeleteIcon from '@material-ui/icons/Delete';
import { setCopyHistoricalBookings, getHistoricalBookingsByUser, getCurrentsBookingsByUser, setOpenModalCancel } from '../../Actions/bookingActions';
import ListBookingStudent from './ListBookingStudent';

const useStyles = makeStyles((theme) => ({
    root: {
        display: 'flex',
        flexFlow: 'row',
        marginTop: '16px',
    },
    title: {
        margin: '15px',
    },
    iconUser: {
        width: '5rem',
        height: '5rem',
        color: '#0000005c',
        margin: '18px 18px 0px 18px',
    },
    avatar: {
        width: '6rem',
        height: '6rem',
        color: '#d5d5d5',
    },
    containerBooking: {
        display: 'flex',
        flexFlow: 'row',
        border: '1px solid #0000002b',
        borderRadius: '4px',
        alignItems: 'center',
        justifyContent: 'space-between',
        height: '2.5rem',
        padding: '0px 12px',
    },
    containerUser: {
        display: 'flex',
        flexFlow: 'column',
        alignItems: 'center',
        justifyContent: 'center',
    },
    containerIconUser: {
        margin: '12px 0px',
    },
    cardUser: {
        margin: '16px 12px 12px 0px',
    },
    cardBookings: {
        marginBottom: '16px',
    },
    nameUser: {
        margin: '12px 0px',
    },
  }));

const Student = ({
    userReducer: {
        user,
    },
    bookingReducer: {
        bookingsHistoricalByUser,
        copyHistoricalBookings,
        bookingsCurrentsByUser,
    },
    setCopyHistoricalBookings,
    getHistoricalBookingsByUser,
    getCurrentsBookingsByUser,
    setOpenModalCancel

}) => {

    const classes = useStyles();

    useEffect(() => {
       getHistoricalBookingsByUser(user.id);
       getCurrentsBookingsByUser(user.id);
    }, [])

    const handleOpenModal = () => {
        setOpenModalCancel(true);
    }

    return (

        <Container maxWidth="lg"> 
            <Grid container spacing={3} className={classes.root} justify="center" >
                <Grid item xs= {12} sm={3} className={classes.cardUser}>

                    <Card>
                        <CardContent>
                            <Grid container className={classes.containerUser}> 
                                <Avatar className={classes.avatar}>
                                    <Grid item className={classes.containerIconUser}>
                                        <FaUser className={classes.iconUser}/>
                                    </Grid>
                                </Avatar>
                                <Typography variant='h9' className={classes.nameUser}><strong>{user.name}</strong></Typography>
                                <Typography variant='body2'>{user.mail}</Typography>
                        </Grid>
                        </CardContent>
                    </Card>
                    
                </Grid>
                <Grid item xs={12} sm={9}>
                    <Grid item xs={12}>
                        <Typography variant='h5' className={classes.title}> <strong>Reservas registradas</strong></Typography>
                    </Grid>
                    <Card className={classes.cardBookings}>
                        <CardContent>
                            <Grid container spacing={3} xs={12} sm={12} className={classes.containetPicture}>
                                <Grid item xs={12} sm={12}>
                                    {bookingsCurrentsByUser.length > 0  ? bookingsCurrentsByUser.map(b =>  
                                        <Grid container row nowrap>
                                            <Grid item xs={11} sm={11} className={classes.containerBooking}>
                                                <Typography variant='body2'>{b.seat.desk.nameDesk}</Typography>
                                                <Typography variant='body2'>Asiento {b.seat.id}</Typography>
                                                <Typography variant='body2'>{moment(b.date).format('LL')}</Typography>
                                                <Typography variant='body2'>{b.startTime}hs - {b.endTime}hs</Typography>
                                            </Grid>
                                            <Grid item xs={1} sm={1} >
                                                <IconButton
                                                aria-label="account of current user"
                                                aria-controls="menu-appbar"
                                                aria-haspopup="true"
                                                onClick={handleOpenModal}
                                                color="primary"
                                                >
                                                    <DeleteIcon />
                                                </IconButton>
                                            </Grid>
                                        </Grid> 
                                        )
                                    : 
                                    <Grid container xs={12} sm={12} justify='center'>
                                        <Typography variant='body2' color='#00000082'>Sin reservas</Typography>
                                    </Grid>
                                    }
                                </Grid>
                            </Grid>
                        </CardContent>
                    </Card>
                    
                    <ListBookingStudent listBooking={bookingsHistoricalByUser} listCopyBooking={copyHistoricalBookings} setCopy={setCopyHistoricalBookings} admin={false}/>

                </Grid>
            </Grid>
        </Container>

    )
}

const mapStateToProps = state => ({
    userReducer: state.userReducer,
    bookingReducer: state.bookingReducer,
});

export default connect(mapStateToProps, { setCopyHistoricalBookings, getHistoricalBookingsByUser, setOpenModalCancel, getCurrentsBookingsByUser  })(Student);