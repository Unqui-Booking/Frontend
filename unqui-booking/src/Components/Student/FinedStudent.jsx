import { Card, CardContent, Typography, Divider, Grid } from '@material-ui/core';
import React from 'react'
import { FaUserLock } from 'react-icons/fa';
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) => ({
    root: {
        height: '80vh',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    title: {
        fontWeight: 'bold',
        textAlign: 'center',
    },
    containerIcon: {
        display: 'flex',
        justifyContent: 'center',
    },
    icon: {
        width: '5rem',
        height: '5rem',
    },
    containetText: {
        display: 'flex',
        alignItems: 'center',
    },
    containerIconText: {
        padding: '20px 10px',
    }

}))


const FinedStudent = (props) => {

    const classes = useStyles();

    return (
        <Grid className={classes.root}>
            <Card>
            <CardContent>
                <Typography variant='h6' className={classes.title}>Usuario multado</Typography>
                <Divider></Divider>
                <Grid container spacing={2} className={classes.containerIconText}>
                    <Grid item xs={3} sm={3} className={classes.containerIcon}>
                        <FaUserLock className={classes.icon}/>
                    </Grid>
                    <Grid item xs={9} sm={9}>
                        <Typography>
                            Tu usuario se encuentra limitado para realizar reservas hasta el día <strong>{props.dateLimit}</strong> inclusive.<br/>
                            En tu perfil podes acceder normalmente a la información detallada allí.
                        </Typography> 
                    </Grid>
                </Grid>
            </CardContent>     
        </Card>
        </Grid>
        
    )
}

export default FinedStudent;