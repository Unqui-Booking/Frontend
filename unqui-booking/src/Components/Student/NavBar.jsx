import React from 'react';
import { connect } from 'react-redux';
import { useHistory } from "react-router-dom";
import { makeStyles } from '@material-ui/core/styles';
import { IconButton, AppBar, Toolbar, Typography } from  '@material-ui/core';
import { AccountCircle } from '@material-ui/icons';
import ExitToAppIcon from '@material-ui/icons/ExitToApp';
import { setFailedLogin } from '../../Actions/userActions';

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  title: {
    flexGrow: 1,
  },
}));

const NavBar = ({
  userReducer: {
    user
  },
  setFailedLogin,

}) => {

  const classes = useStyles();
  let history = useHistory(); 

  const handleLogOut = () => {
   
      setFailedLogin(false);
      history.push("/");
    
    
  }

  const handleProfile = async () => {
      history.push("/student");
  }
  
    return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" className={classes.title}>
            UNQui-Booking
          </Typography>
          
          {user != undefined ?
            <div>
              { !user.admin ?
                <IconButton
                  aria-label="account of current user"
                  aria-controls="menu-appbar"
                  aria-haspopup="true"
                  onClick={handleProfile}
                  color="inherit"
                >
                  <AccountCircle />
                </IconButton>: null 
              }
              <IconButton
                aria-label="account of current user"
                aria-controls="menu-appbar"
                aria-haspopup="true"
                onClick={handleLogOut}
                color="inherit"
              >
                <ExitToAppIcon/>
              </IconButton>
            </div> : null
          }
          
        </Toolbar>
      </AppBar>
    </div>
  );

}

const mapStateToProps = state => ({
  userReducer: state.userReducer,
  bookingReducer: state.bookingReducer,
});

export default connect(mapStateToProps, { setFailedLogin })(NavBar)