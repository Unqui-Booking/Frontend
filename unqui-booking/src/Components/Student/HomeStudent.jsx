import React, { useState } from 'react'
import { connect } from 'react-redux';
import moment from 'moment';
import { Grid, Container, Typography} from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import { MuiPickersUtilsProvider, KeyboardDatePicker } from '@material-ui/pickers';
import SelectPlace from './SelectPlace/SelectPlace';
import BookingRegister from './BookingRegister';
import { setSelectedDate } from  '../../Actions/dateHoursActions';
import { getMapAvailabilySeats } from '../../Actions/bookingActions';
import { isFinedUserAtDate } from '../../Actions/userActions';
import { useEffect } from 'react';
import FinedStudent from '../Student/FinedStudent';
import CircularProgress from '@material-ui/core/CircularProgress';
import DateFnsUtils from '@date-io/date-fns';
import deLocale from "date-fns/locale/es";


const useStyles = makeStyles((theme) => ({
    formControl: {
        margin: theme.spacing(1),
        minWidth: 120,
      },
      selectEmpty: {
        marginTop: theme.spacing(2),
      },
      flex: {
          display: "flex",
      },
      title: {
        padding: "2rem",
        textAlign: "center",
      },
      area: {
          backgroundColor: "rgba(0, 0, 0, 0.1)",
          border: "6px",
          borderStyle: "outset",
          paddingTop: "1rem",
      },
      dateSelect: {
          marginBottom: "2rem",
      },

}))

const HomeStudent = ({

    dateHoursReducer: {
        date,
        startTime,
        endTime
    },
    alertMessageReducer: {
        activeStep,
    },
    deskReducer: {
        desk
    },
    userReducer: {
        user,
    },
    setSelectedDate,
    getMapAvailabilySeats,
    isFinedUserAtDate

    }) => {

    const classes = useStyles();
    const [finedUser, setFinedUser] = useState();
    const [dateLimit, setDateLimit] = useState();
    const [loading, setLoading] = useState(true);

    useEffect(()=> {
        setInfoFined(user.id);
    }, [])

    const setInfoFined = async (userId) => {
        let infoFined = await isFinedUserAtDate(moment(new Date()).format().split('T')[0], userId);
        let isFined =  infoFined.fined;
        let dateLimit =  infoFined.dateLimit;
        setFinedUser(isFined);
        setDateLimit(dateLimit);
        setLoading(false);
    } 

    const filterDays = (date) => {
        const today = new Date();
        //|| date.getDay() === 6 
        return date.getDay() === 0 || date.getDate() > (today.getDate() + 14);
      }
    
    const handleOnChange = (date) => {
        setSelectedDate(date);
        if(desk != null){ //si selecciono el cambio de fecha en el paso 2
            getMapAvailabilySeats(desk.id, moment(date).format().split('T')[0], startTime, endTime);
        }
    }
  
    return (
        loading ? <CircularProgress size={24} className={classes.buttonProgress} /> :
        <Container maxWidth="md">    
            {finedUser ? 
                <FinedStudent dateLimit={dateLimit}/> :
                <Grid container className={classes.root} justifyContent="center" > 
                    <Grid item xs={12}>
                        <Typography variant='h4' data-testid='title-home-student' className={classes.title}> Sistema de gesti√≥n de reservas de escritorios</Typography>
                    </Grid>
                    <Grid item xs={6} justifyContent="center" className={classes.flex}>
                        <MuiPickersUtilsProvider locale={deLocale} utils={DateFnsUtils}  >
                            <KeyboardDatePicker
                                disabled={ activeStep == 2  }
                                className={classes.dateSelect}
                                disableToolbar
                                variant="inline"
                                format="dd/MM/yyyy"
                                margin="normal"
                                id="date-picker-inline"
                                value={date}
                                label="Seleccionar fecha de reserva"
                                KeyboardButtonProps={{
                                    'aria-label': 'change date',
                                }}
                                onChange={handleOnChange}
                                disablePast={true}
                                shouldDisableDate={filterDays}
                            />
                        </MuiPickersUtilsProvider>
                    </Grid>
                    <Grid item xs={6} justifyContent="center" className={classes.flex}>
                        <BookingRegister/>
                    </Grid>
                    <Grid item xs={12}>
                        <SelectPlace/>
                    </Grid>
                </Grid>
            }
        </Container>  
    )}

const mapStateToProps = state => ({
    dateHoursReducer: state.dateHoursReducer,
    alertMessageReducer: state.alertMessageReducer,
    deskReducer: state.deskReducer,
    userReducer: state.userReducer,
});
    
export default connect(mapStateToProps, { setSelectedDate, getMapAvailabilySeats, isFinedUserAtDate })(HomeStudent);