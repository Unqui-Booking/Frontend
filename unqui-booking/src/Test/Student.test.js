import React from 'react'
import { render, screen } from '@testing-library/react'
import Student from '../Components/ProfileStudent/Student'
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);


describe('Student', () => {

  let store;
  let component;
 
  beforeEach(() => {
    
    store = mockStore({
      userReducer: { user: {
                "id": 1,
                "name": "Erica Gerez",
                "mail": "erica.gerez@alu.edu.unq.com.ar",
                "password": "Contrasenia1",
                "admin": false,
                "deleted": false
            },
    },
      bookingReducer: {
        bookingsHistoricalByUser: [],
        bookingsCurrentsByUser: [],
        copyHistoricalBookings: [],
        openModalCancel: false
      }
   
  });

    component = renderer.create(
      <Provider store={store}>
        <Student />
      </Provider>
    );

  });

  it('when user is logged, show name user', async () => {
    //console.log(component.toJSON());
    expect(component.toJSON()).toMatchSnapshot();
    //expect(component.toJSON().queryByText(/registradas/i)).toBeInTheDocument();
    // expect(screen.queryByText(/registradas/i)).toBeInTheDocument();
   
    // const renderComponent = (user ) =>
    //   render(
    //   <Provider store={createStore(reducerUser, { user })}>
    //       <Student />
    //   </Provider>
    //   );

    // renderComponent(
    //     { user: {
    //         "id": 1,
    //         "name": "Erica Gerez",
    //         "mail": "erica.gerez@alu.edu.unq.com.ar",
    //         "password": "Contrasenia1",
    //         "admin": false,
    //         "deleted": false
    //     } });

     // expect(screen.queryByText(/erica gerez/i)).toBeInTheDocument();
       //expect(page).toHaveTextContent(/erica gerez/i)

      //expect(getByTestId('name-user')).toBeInTheDocument()
    

   

  });



})