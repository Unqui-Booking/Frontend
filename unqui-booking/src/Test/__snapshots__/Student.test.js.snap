// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student when user is logged, show name user 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-content-xs-center"
  >
    <div
      className="MuiGrid-root makeStyles-cardUser-8 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardContent-root"
        >
          <div
            className="MuiGrid-root makeStyles-containerUser-6 MuiGrid-container"
          >
            <div
              className="MuiAvatar-root MuiAvatar-circular makeStyles-avatar-4 MuiAvatar-colorDefault"
            >
              <div
                className="MuiGrid-root makeStyles-containerIconUser-7 MuiGrid-item"
              >
                <svg
                  className="makeStyles-iconUser-3"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                  />
                </svg>
              </div>
            </div>
            <h6
              className="MuiTypography-root makeStyles-nameUser-10 MuiTypography-h6"
              data-testid="name-user"
            >
              <strong>
                Erica Gerez
              </strong>
            </h6>
            <p
              className="MuiTypography-root MuiTypography-body2"
            >
              erica.gerez@alu.edu.unq.com.ar
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h5
          className="MuiTypography-root makeStyles-title-2 MuiTypography-h5"
        >
           
          <strong>
            Reservas registradas
          </strong>
        </h5>
      </div>
      <div
        className="MuiPaper-root MuiCard-root makeStyles-cardBookings-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardContent-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
          >
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
            >
              <div
                className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
              >
                <p
                  className="MuiTypography-root MuiTypography-body2"
                >
                  Sin reservas
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-containerFilter-19 MuiGrid-container"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
          >
            <div
              className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
                htmlFor="filterDate"
                id="filterDate-label"
                style={
                  Object {
                    "fontSize": 15,
                  }
                }
              >
                Filtrar por fecha
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl MuiInputBase-adornedEnd"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd"
                  disabled={false}
                  id="filterDate"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
                <div
                  className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                >
                  <button
                    aria-label="change date"
                    className="MuiButtonBase-root MuiIconButton-root"
                    disabled={false}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    tabIndex={0}
                    type="button"
                  >
                    <span
                      className="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                        />
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-marginTop-14 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
          >
            <div
              className="MuiFormControl-root MuiTextField-root makeStyles-width-16"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
                htmlFor="filterdDesk"
                id="filterdDesk-label"
                style={
                  Object {
                    "fontSize": 15,
                  }
                }
              >
                Filtrar por escritorio
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
                shrink={true}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  id="filterdDesk"
                  max={null}
                  min={1}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="number"
                  value={null}
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-marginTop-14 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
          >
            <div
              className="MuiFormControl-root MuiTextField-root makeStyles-width-16"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink={false}
                htmlFor="filterdSeat"
                id="filterdSeat-label"
                style={
                  Object {
                    "fontSize": 15,
                  }
                }
              >
                Filtrar por asiento
              </label>
              <div
                className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                onClick={[Function]}
                shrink={true}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiInput-input"
                  disabled={false}
                  id="filterdSeat"
                  max={null}
                  min={1}
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="number"
                  value={null}
                />
              </div>
            </div>
          </div>
          <div
            className="MuiGrid-root makeStyles-clearFilter-18 MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-1"
          >
            <button
              aria-haspopup="true"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 13h14v-2H5v2zm-2 4h14v-2H3v2zM7 7v2h14V7H7z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-accordion-12 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
        >
          <div
            className="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              aria-controls="panel1a-content"
              aria-disabled={false}
              aria-expanded={true}
              className="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
              id="panel1a-header"
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiAccordionSummary-content Mui-expanded"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1"
                >
                  <strong>
                    Históricos
                  </strong>
                </p>
              </div>
              <div
                aria-disabled={false}
                aria-hidden={true}
                className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role={null}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
              </div>
            </div>
            <div
              className="MuiCollapse-root MuiCollapse-entered"
              style={
                Object {
                  "minHeight": "0px",
                }
              }
            >
              <div
                className="MuiCollapse-wrapper"
              >
                <div
                  className="MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby="panel1a-header"
                    id="panel1a-content"
                    role="region"
                  >
                    <div
                      className="MuiAccordionDetails-root makeStyles-accordionDetails-13"
                    >
                      <div
                        className="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-center"
                      >
                        <p
                          className="MuiTypography-root MuiTypography-body2"
                        >
                          Sin reservas
                        </p>
                      </div>
                      <nav
                        aria-label="pagination navigation"
                        className="MuiPagination-root makeStyles-pagination-17"
                      >
                        <ul
                          className="MuiPagination-ul"
                        >
                          <li>
                            <button
                              aria-label="Go to first page"
                              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-sizeSmall Mui-disabled"
                              disabled={true}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={-1}
                              type="button"
                            >
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiPaginationItem-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                                />
                              </svg>
                            </button>
                          </li>
                          <li>
                            <button
                              aria-label="Go to previous page"
                              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-sizeSmall Mui-disabled"
                              disabled={true}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={-1}
                              type="button"
                            >
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiPaginationItem-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                                />
                              </svg>
                            </button>
                          </li>
                          <li>
                            <button
                              aria-label="Go to next page"
                              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-sizeSmall Mui-disabled"
                              disabled={true}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={-1}
                              type="button"
                            >
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiPaginationItem-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                                />
                              </svg>
                            </button>
                          </li>
                          <li>
                            <button
                              aria-label="Go to last page"
                              className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-sizeSmall Mui-disabled"
                              disabled={true}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={-1}
                              type="button"
                            >
                              <svg
                                aria-hidden={true}
                                className="MuiSvgIcon-root MuiPaginationItem-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                                />
                              </svg>
                            </button>
                          </li>
                        </ul>
                      </nav>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
